### PR TITLE
Improvement of SID and plot functions

### PR DESCRIPTION
**Pull request by dringeis**
- Faster plot: transform outside of tripcolor
 - Config.py as now a function returning a dictionnary
 - the config dict is given to the functions
 - remove id_start_latX , keeping idX
 - Move function in a file called utils.py for functions that have nothing to do with utils
 - Save timeseries data in a pickle file
 - cleanup of the loading packages
 - progress bars in most of the for loops

### CODE REVIEW

**Review by dringeis — COMMENTED** [review]
Just ran a check through everything, looks good to me

File: src/SatelliteCoverage/config.py [hunk 1]
```diff
@@ -75,44 +74,17 @@ def compile_data(raw_paths):
...
-    out_proj = pyproj.Proj('+proj=stere +lat_0=90 +lat_ts=70 +lon_0=0 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs ', preserve_units=True)
-
-    # Transform function
-    x, y = np.array(pyproj.transform(in_proj, out_proj, lon, lat))
-    return x, y
-
 # Divides raw data into intervals specified by the user
 # def divide_intervals(raw_paths, max_date, min_date, interval):
-def divide_intervals(raw_paths, Date_options = None, Options = None):
+def divide_intervals(config):
```
Comment (a-bouchat) [inline src/SatelliteCoverage/config.py:87]:
```suggestion
def divide_intervals(config=None):
```

File: src/SatelliteCoverage/netcdf_tools.py [hunk 2]
```diff
@@ -116,8 +129,211 @@ def plot_start_end_points(path:str):
             os.remove(fig_path)
 
     # Saving figure
+    print('Saving start and end points figure at ',fig_path)
     plt.savefig(fig_path, bbox_inches='tight', dpi=600)
 
+# Filters netCDF data by area
+def filter_area(centre_lat, centre_lon, radius, start_lats, start_lons):
```
Comment (a-bouchat) [inline src/SatelliteCoverage/netcdf_tools.py:135]: Change to config

File: src/SatelliteCoverage/netcdf_tools.py [hunk 3]
```diff
@@ -203,48 +428,67 @@ def plot_deformations(path:str):
...
+                                                        data['idx2'][min_index:max_index],
+                                                        data['idx3'][min_index:max_index])
 
         # Extracting deformation data
         div_colours = data['div'][min_index:max_index]
         shr_colours = data['shr'][min_index:max_index]
         vrt_colours = data['vrt'][min_index:max_index]
 
+        # tranform the coordinates already to improve the plot efficiency
+        new_coords = ax_div.projection.transform_points(trans, np.array(start_lon), np.array(start_lat))
```
Comment (a-bouchat) [inline src/SatelliteCoverage/netcdf_tools.py:462]:
```suggestion
        new_coords = proj.transform_points(trans, np.array(start_lon), np.array(start_lat))
```

File: src/SeaIceDeformation/visualise_deformation.py [hunk 4]
```diff
@@ -91,11 +93,17 @@ def visualise_deformations():
         # Stack the 3 vertices index arrays
         triangles = np.stack((vertice_idx1, vertice_idx2, vertice_idx3), axis=-1)
 
+        # tranform the coordinates already to improve the plot efficiency
+        new_coords = ax_div.projection.transform_points(trans, np.array(s_lon), np.array(s_lat))
```
Comment (a-bouchat) [inline src/SeaIceDeformation/visualise_deformation.py:97]:
```suggestion
        new_coords = proj.transform_points(trans, np.array(s_lon), np.array(s_lat))
```

File: src/SatelliteCoverage/netcdf_tools.py (hunk 2 above)
Comment (dringeis) [reply src/SatelliteCoverage/netcdf_tools.py]: done

**Comment by a-bouchat** [comment]
Beautiful. Approved.

**Pull request merged**